src/modules/finance/index.ts

```diff
@@ -486,23 +490,44 @@ export class FinanceModule extends ModuleBase {
   /**
    * Generates a random Bitcoin address.
    *
+   * @param options An options object.
+   * @param options.type The bitcoin address type (legacy, sewgit, bech32 or taproot). Defaults to `legacy`
+   * @param options.network The bitcoin network (mainnet or testnet). Defaults to `mainnet`.
+   *
    * @example
-   * faker.finance.bitcoinAddress() // '3ySdvCkTLVy7gKD4j6JfSaf5d'
+   * faker.finance.bitcoinAddress() // '1TeZEFLmGPLEQrSRdAcnZLoWwYeiHwmRog'
+   * faker.finance.bitcoinAddress({ type: 'bech32' }) // 'bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t4'
+   * faker.finance.bitcoinAddress({ type: 'bech32', network: 'testnet' }) // 'tb1qw508d6qejxtdg4y5r3zarvary0c5xw7kxpjzsx'
    *
    * @since 3.1.0
    */
-  bitcoinAddress(): string {
-    const addressLength = this.faker.number.int({ min: 25, max: 39 });
-
-    let address = this.faker.helpers.arrayElement(['1', '3']);
-
-    address += this.faker.string.alphanumeric({
-      length: addressLength,
-      casing: 'mixed',
-      exclude: '0OIl',
+  bitcoinAddress(options?: {
+    /**
+     * The bitcoin address type (legacy, sewgit, bech32 or taproot). Defaults to `legacy`.
+     *
+     * @default 'legacy'
+     */
+    type?: BitcoinAddressType;
+    /**
+     * The bitcoin network (mainnet or testnet). Defaults to `mainnet`.
+     *
+     * @default 0
+     */
+    network?: BitcoinNetwork;
+  }): string {
+    const { type = 'legacy', network = 'mainnet' } = options || {};
```

Comment:
We usually default to `{}` in the parameters.

```suggestion
  } = {}): string {
    const { type = 'legacy', network = 'mainnet' } = options;
```